roachtest/sqlsmith: num_inverted_index_entries with a NULL argument internal error

roachtest.sqlsmith/setup=rand-tables/setting=no-ddl [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5312207?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/5312207?buildTab=artifacts#/sqlsmith/setup=rand-tables/setting=no-ddl) on master @ [e6815947a050e32f21e983aa30dc74ab2a247af3](https://github.com/cockroachdb/cockroach/commits/e6815947a050e32f21e983aa30dc74ab2a247af3):


```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /artifacts/sqlsmith/setup=rand-tables/setting=no-ddl/run_1
	sqlsmith.go:265,sqlsmith.go:305,test_runner.go:884: error: pq: internal error: expected *DString, found tree.dNull
		stmt:
		WITH
			with_26891 (col_154322)
				AS (
					SELECT
						*
					FROM
						(
							VALUES
								('\x52b558209d':::BYTES),
								('\xeaadc110ec':::BYTES),
								(
									set_bit('\x029881384aff':::BYTES::BYTES, 1697719185248289085:::INT8::INT8, crdb_internal.num_inverted_index_entries(NULL::STRING, (-4539479078906640300):::INT8::INT8)::INT8::INT8)::BYTES
								),
								(
									(crdb_internal.replication_stream_spec((SELECT bit_or((-4525244141147675471):::INT8::INT8)::INT8 AS col_154321 FROM defaultdb.public.table1@table1_pkey AS tab_80677, defaultdb.public.table2@[0] AS tab_80678 WHERE true GROUP BY tab_80678.col2_7 HAVING every(false::BOOL)::BOOL LIMIT 1:::INT8)::INT8)::BYTES::BYTES || '\x1d3fb29196dbc0':::BYTES::BYTES)::BYTES
								),
								('\x9c':::BYTES)
						)
							AS tab_80679 (col_154322)
				)
		SELECT
			cte_ref_7895.col_154322 AS col_154323, '\x84d511688b34b66539':::BYTES AS col_154324
		FROM
			with_26891 AS cte_ref_7894, with_26891 AS cte_ref_7895
		LIMIT
			38:::INT8;
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*sqlsmith/setup=rand-tables/setting=no-ddl.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-16191

--- a/pkg/sql/logictest/testdata/logic_test/crdb_internal
+++ b/pkg/sql/logictest/testdata/logic_test/crdb_internal
@@ -1136,3 +1136,10 @@ SELECT * FROM t69684 WHERE crdb_internal.increment_feature_counter(a)
 # The output is non-deterministic.
 statement ok
 SELECT crdb_internal.probe_ranges(INTERVAL '1000ms', 'write')
+
+# Regression test for not handling NULL values correctly by an internal builtin
+# (#82056).
+query I
+SELECT crdb_internal.num_inverted_index_entries(NULL::STRING, 0)
+----
+0

--- a/pkg/sql/sem/builtins/builtins.go
+++ b/pkg/sql/sem/builtins/builtins.go
@@ -5668,6 +5668,9 @@ value if you rely on the HLC for accuracy.`,
 				// TODO(jordan): need to re-evaluate when we support more than just
 				// trigram inverted indexes.
 				// The version argument is currently ignored for string inverted indexes.
+				if args[0] == tree.DNull {
+					return tree.DZero, nil
+				}
 				s := string(tree.MustBeDString(args[0]))
 				return tree.NewDInt(tree.DInt(len(trigram.MakeTrigrams(s, true /* pad */)))), nil
 			},

